include/exadg/aero_acoustic/single_field_solvers/fluid.h

```diff
@@ -86,12 +90,58 @@ class SolverFluid
                 dealii::ExcMessage("Invalid parameter in context of fluid-structure interaction."));
 
     // initialize time_integrator
-    time_integrator = IncNS::create_time_integrator<dim, Number>(pde_operator,
-                                                                 nullptr /*no ALE*/,
-                                                                 postprocessor,
-                                                                 application->get_parameters(),
-                                                                 mpi_comm,
-                                                                 is_test);
+    if(compute_acoustic_from_analytical_cfd_solution)
+    {
+      auto interpolate_analytical_solution =
+        [&, analytical_cfd_solution_velocity, analytical_cfd_solution_pressure](
+          VectorType & velocity, VectorType & pressure, double const time) {
+          velocity.zero_out_ghost_values();
+          pressure.zero_out_ghost_values();
+
+          analytical_cfd_solution_velocity->set_time(time);
+          analytical_cfd_solution_pressure->set_time(time);
+
+          // This is necessary if Number == float
+          using VectorTypeDouble = dealii::LinearAlgebra::distributed::Vector<double>;
+
+          VectorTypeDouble velocity_double;
+          VectorTypeDouble pressure_double;
+          velocity_double = velocity;
+          pressure_double = pressure;
+
+          dealii::VectorTools::interpolate(*pde_operator->get_mapping(),
+                                           pde_operator->get_dof_handler_u(),
+                                           *(analytical_cfd_solution_velocity),
+                                           velocity_double);
+
+          dealii::VectorTools::interpolate(*pde_operator->get_mapping(),
+                                           pde_operator->get_dof_handler_p(),
+                                           *(analytical_cfd_solution_pressure),
+                                           pressure_double);
+
+          velocity = velocity_double;
+          pressure = pressure_double;
+        };
```

Comment:
I think this should be functionality of `IncNS::SpatialOperatorBase`, similar to `IncNS::SpatialOperatorBase::prescribe_initial_conditions()`. `IncNS::FieldFunctions` already has `analytical_solution_pressure`, so one would have to add `analytical_solution_velocity`.

Reply:
[b7860ce](https://github.com/exadg/exadg/pull/659/commits/b7860ce30c3cbaeee80a8cfc4f1a4c3247d59845)